drgn/commands/_builtin/crash/ps.py

```diff
@@ -0,0 +1,294 @@
+            "task_vsize",
+            "get_task_rss_info",
+            "totalram_pages",
+        )
+        builder.add_from_import(
+            "drgn.helpers.linux.sched", "task_state_to_char", "task_cpu"
+        )
+        if args.last_run or args.last_run_timestamp:
+            builder.add_from_import(
+                "drgn.helpers.linux.sched", "task_since_last_arrival_ns"
+            )
+            builder.add_import("datetime")
+
+        builder.append("total_mem = int(totalram_pages(prog))\n")
+        builder.append("page_size = prog['PAGE_SIZE'].value_()\n")
+
+        if args.user or args.kernel:
+            builder.add_from_import("drgn.helpers.linux.kthread", "is_kthread")
+            if args.user:
+                builder.append(
+                    "tasks = filter(lambda t: not task_is_kthread(t), for_each_task(prog))\n"
+                )
+            elif args.kernel:
+                builder.append("tasks = filter(task_is_kthread, for_each_task(prog))\n")
+        else:
+            builder.append("tasks = for_each_task(prog)\n")
+
+        if args.group_leader:
+            builder.add_from_import(
+                "drgn.helpers.linux.sched",
+                "thread_group_leader",
+            )
+            builder.append("tasks = filter(thread_group_leader, tasks)\n")
+
+        builder.append("for task in tasks:\n")
+        builder.append("    cpu = task_cpu(task)\n")
+        builder.append("    state = task_state_to_char(task)\n")
+        builder.append("    task_rss = get_task_rss_info(task)\n")
+        builder.append("    total_rss = task_rss.total\n")
+        builder.append("    pct_mem = total_rss * 100 / total_mem\n")
+        builder.append("    vmem = task_vsize(task)\n")
+        builder.append("    command = escape_ascii_string(task.comm.string_())\n")
+
+        if args.last_run:
+            builder.append("    time_nanosec = task_since_last_arrival_ns(task)\n")
+            builder.append(
+                "    last_arrival = format_nanosecond_duration(time_nanosec)\n"
+            )
+        elif args.last_run_timestamp:
+            builder.append("    last_arrival = task_since_last_arrival_ns(task)\n")
+
+        builder.print()
+        return
+    tasks = list(for_each_task(prog))
+    if args.arg is not None:
+        if args.arg[0] == "pid":
+            task = find_task(prog, args.arg[1])
+            tasks = [task] if task is not None else []
+        elif args.arg[0] == "task":
+            tasks = [Object(prog, "struct task_struct *", value=args.arg[1])]
+        elif args.arg[0] == "comm":
+            tasks = [
+                t for t in tasks if escape_ascii_string(t.comm.string_()) == args.arg[1]
+            ]
+    if args.user:
+        tasks = sorted(filter(lambda t: not task_is_kthread(t), tasks))
+    elif args.kernel:
+        tasks = sorted(filter(task_is_kthread, tasks))
+    if args.group_leader:
+        tasks = list(filter(thread_group_leader, tasks))
```

Comment:
We don't need to sort here yet since the `show_*` functions already sort, right?

Also, list comprehensions are more Pythonic (and probably slightly more performant than `list(filter(...))`.

```suggestion
    if args.user:
        tasks = [task for task in tasks if not task_is_kthread(t)]
    elif args.kernel:
        tasks = [task for task in tasks if task_is_kthread(t)]
    if args.group_leader:
        tasks = [task for task in tasks if thread_group_leader(t)]
```